### ISSUE TITLE
Increasing scores while training a deep autoencoder with RBM

### ISSUE THREAD

**Issue by DesmondYuan** [deeplearning4j/deeplearning4j#2190]
I am training a deep autoencoder 978->500->250->100->30 but it seems quite weird that during the pretraining phase the score (both for RMSE_XENT and MSE )is increasing.  Changing random seeds does not help. Adding some regularization or gradient normalization method does not help that much, only slow down the increase rate. I've tried a batchSize=100 and 1000, does not help...

A piece of log and the full code can be found here https://gist.github.com/DesmondYuan/159b8f0064e9375801810d099e1e14e6

**Comment by DesmondYuan** [comment]
changing the LossFunction from XENT to MSE for outputLayer stops the bleeding and the finetune score is falling now. But the remains increasing, though slowly.

**Comment by DesmondYuan** [comment]
gotta copy the link and open it in the browser. dont know why but simple clicking the link leads to 404.

**Comment by treo** [comment]
@nyghtowl Here are some more very simple examples as promised yesterday:
https://gist.github.com/treo/bfa2289c24b25328ea7dc17fa0701ead

I'm using cosine proximity as the loss function there as I find it to be easier to interpret, and MSE results in the same behavior.

Also notice how disabling pretraining on Auto Encoders makes them behave exactly the same as the simple mlp but doing the same for RBMs still doesn't allow them to get quite the same result even when initialized with the correct solution.

**Comment by nyghtowl** [comment]
Thanks @treo & @DesmondYuan

**Comment by hitkang** [comment]
Hi @nyghtowl , is there some update regarding this issue. I met the the same problem that the score is not decreasing for AutoEncoder and RBM.

**Comment by vlovsky** [comment]
Please check again on 0.8.1 build. I fixed two issues that most likely were involved in this.

**Comment by eraly** [comment]
We are dropping support for RBMs in favour of newer and more effective methods. Please take a look at variational autoencoders 
https://github.com/deeplearning4j/dl4j-examples/tree/master/dl4j-examples/src/main/java/org/deeplearning4j/examples/unsupervised/variational
